### ISSUE TITLE
Crashes Actual Craft Time

### ISSUE THREAD

**Issue by eihns** [jpanther/Dectorio#97]
**Describe the bug**
I have many many many mods installed, but only if i enable your mod, i get the following crash on actual craft time:
https://mods.factorio.com/mod/Actual_Craft_Time/discussion/5c92b03bcc0838000cf6c0ef

**How to reproduce**
Install Actual Craft time
Install Dectorio
Go into game and click any random building. Crash.

**Comment by jpanther** [comment]
Thanks looking at this now. I'm able to replicate it.

```
The mod Actual Craft Time caused a non-recoverable error.

Error while running event Actual_Craft_Time::on_gui_opened (ID 83)
__Actual_Craft_Time__/control.lua:148: attempt to perform arithmetic on field 'belt_speed' (a nil value)
stack traceback:
	__Actual_Craft_Time__/control.lua:148: in function 'findPrototypeData'
	__Actual_Craft_Time__/control.lua:434: in function <__Actual_Craft_Time__/control.lua:351>```

**Comment by jpanther** [comment]
Okay, I have worked out the cause of the issue. It's something that needs to be fixed by the other mod developer.

Basically, the _Actual_Craft_Time_ mod is creating copies of any entities that contain the certain keywords in their name - `inserter`, `filter`, etc. Meanwhile, Dectorio is creating icon entities for in-game entities that are based upon the original entity name - for instance, an entity called `inserter` gets an icon (`type=simple-entity`) called `dect-icon-inserter`. Because the _Actual_Craft_Time_ mod is not applying any additional checks on the entities other than using their name, there is a potential for this sort of incompatibility as it expects there to be a `belt_speed` property on these entities, and the icon prototypes do not have this property.

In short, the fix is for _Actual_Craft_Time_ to test the entities have the required `belt_speed` property before making a copy of the entity. That way, they will only ever copy entities that are relevant and avoid any incompatibilities with other mods. In the meantime, you can workaround the incompatibility by changing your Mod Settings and disabling the "Enable Signs" startup option in Dectorio.

**Comment by eihns** [comment]
Thanks, ill pass this to the dev of the other mod! Amazing fast work, wtf!

**Comment by osldgoth** [comment]
Would you tell me where in your code you are copying inserters(entity and/or item)? Part of the problem is your code makes a copy of the inserters after mine. So your code copies my copies which causes issues later in my code. As far as I can tell*

**Comment by jpanther** [comment]
It happens during the data-final-fixes stage (as it needs to create icons for any items added by other mods). You can see the logic here `Dectorio/prototypes/entity/signs-icons.lua`. It will essentially create an icon `simple-entity` for any prototype that is of one of these types `{"item", "fluid", "tool", "ammo", "armor", "capsule", "gun", "module"}`.

**Comment by jpanther** [comment]
Just tested this with the 0.3.7 update for _Actual Craft Time_ and it seems to have fixed the issue. Thanks @osldgoth.
